ooniprobe/Utility/TestUtility.mm

```diff
@@ -152,4 +131,16 @@ + (NSString*)getFileContent:(NSString*)fileName{
     return nil;
 }
 
++ (void)writeString:(NSString*)str toFile:(NSString*)fileName{
+    NSFileHandle *fileHandle = [NSFileHandle fileHandleForWritingAtPath:fileName];
+    if (fileHandle){
+        [fileHandle seekToEndOfFile];
+        [fileHandle writeData:[[NSString stringWithFormat:@"\n%@", str] dataUsingEncoding:NSUTF8StringEncoding]];
+        [fileHandle closeFile];
+    }
+    else {
```

Comment:
I don't understand very well this algorithm: how do you know that, if `fileHandle` is `nil` then it's necessarily because the `fileHandle` does not exist? Couldn't this error condition also be caused by other errors?

Reply:
The `if (fileHandle)` code cheks for nil
the doc for `fileHandleForWritingAtPath` states


The file pointer is set to the beginning of the file. You cannot read data from the returned file handle object. Use the writeData: method to write data to the file handle. When using this method to create a file handle object, the file handle owns its associated file descriptor and is responsible for closing it.
--
 